library/gcse/src/gcse_query_parameters.e

```diff
@@ -0,0 +1,237 @@
+note
+	description: "[
+		Represent google custom search parameters
+		Example url template
+		  "template": "https://www.googleapis.com/customsearch/v1?q={searchTerms}&num={count?}&start={startIndex?}&lr={language?}&safe={safe?}&cx={cx?}&cref={cref?}&sort={sort?}&filter={filter?}&gl={gl?}&cr={cr?}&googlehost={googleHost?}&c2coff={disableCnTwTranslation?}&hq={hq?}&hl={hl?}&siteSearch={siteSearch?}&siteSearchFilter={siteSearchFilter?}&exactTerms={exactTerms?}&excludeTerms={excludeTerms?}&linkSite={linkSite?}&orTerms={orTerms?}&relatedSite={relatedSite?}&dateRestrict={dateRestrict?}&lowRange={lowRange?}&highRange={highRange?}&searchType={searchType}&fileType={fileType?}&rights={rights?}&imgSize={imgSize?}&imgType={imgType?}&imgColorType={imgColorType?}&imgDominantColor={imgDominantColor?}&alt=json"
+	]"
+	date: "$Date: 2015-10-08 07:51:29 -0300 (ju., 08 oct. 2015) $"
+	revision: "$Revision: 97966 $"
+	EIS: "GCSE parameters", "src=https://developers.google.com/custom-search/json-api/v1/reference/cse/list", "protocol=URI"
+
+class
+	GCSE_QUERY_PARAMETERS
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make (a_secret_key,  a_cx, a_query: READABLE_STRING_8)
+			-- Create an object GCSE_QUERY_PARAMETERS with secret key `a_secret_key' and a custom search engine id `a_cx'.
+			-- and query `a_query'.
+		do
+				-- TODO
+				-- At the moment the API only use cx as Google Custom Search id.
+				-- Custom search engine ID - Use either cx or cref to specify the custom search engine you want to use to perform this search
+
+			secret := a_secret_key
+			cx := a_cx
+			query := a_query
+		ensure
+			secret_set: secret.same_string (a_secret_key)
+			cx_set: cx.same_string (a_cx)
+			query_set: query.same_string (a_query)
+		end
+
+feature -- Access : Required Parameters
+
+	secret: READABLE_STRING_8
+			-- Required. The shared key between your site and Google Custom Search Engine.
+
+	cx: READABLE_STRING_8
+			-- Custom search engine id to perform this search.
+
+	query: READABLE_STRING_8
+			-- Search query, query parameter to specify your search expression.
+
+feature -- Optional Parameters
+
+--Optional parameters
```

Comment:
This should be in an indexing note.

(side note, it may be nice to have indexing note for feature clauses..)